Unit test failures on main

There is at least one, but possibly more unit test failures on main. Most of the code changes in the last 2 months have been committed to main so the unit tests have not been run. 

- PR https://github.com/mantidproject/download.mantidproject.org/pull/33#pullrequestreview-981529875 failed the unit test github action, failing [`test_invalid_codename_throws`](https://github.com/mantidproject/download.mantidproject.org/runs/6553061865?check_suite_focus=true).
- I think this may simply be the first unit test to fail, as running them locally I get the following:

<details>
	<summary>Locally run test output</summary>
		  
============================================= test session starts =============================================
platform linux -- Python 3.6.9, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/danielmurphy/download.mantidproject.org
collected 39 items                                                                                            

tools/tests/test_os_handling.py ......F.........                                                        [ 41%]
tools/tests/test_release_parsing.py ..............F                                                     [ 79%]
tools/tests/test_static_vars.py ...                                                                     [ 87%]
tools/tests/test_url_handling.py .....                                                                  [100%]

================================================== FAILURES ===================================================
________________________________________ test_invalid_codename_throws _________________________________________

    def test_invalid_codename_throws():
        with pytest.raises(KeyError):
>           os_handling.get_osx_codename("Unknown")
E           Failed: DID NOT RAISE <class 'KeyError'>

tools/tests/test_os_handling.py:22: Failed
____________________________________________ test_mantid_releases _____________________________________________

    def test_mantid_releases():
>       found = release_parsing.get_mantid_releases()

tools/tests/test_release_parsing.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tools/impl/release_parsing.py:123: in get_mantid_releases
    releases.extend(_parse_build_names(release_txt_file))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

file_path = PosixPath('/home/danielmurphy/download.mantidproject.org/releases/3.0.0.txt')

    def _parse_build_names(file_path: Path) -> List[ReleaseInfo]:
        """
        Parses a file that contains build names (e.g. those in /releases/) and stores the contents in a dictionary.
        The key of the dictionary is the operating system, which is obtained based on the build's file extension.
        The date on the first line is optional, if it is not present the date is extracted from the first filename if
        possible, else it raises an error.
    
        Args:
          file_path (str): The location of the release file to parse.
    
        Returns:
          List[ReleaseInfo] object containing details about a release
        """
        file_name = file_path.name
        is_nightly = NIGHTLY_NAME_SUFFIX.casefold() in file_name.casefold()
    
        if is_nightly:
            version = "nightly"
        else:
            version = file_path.with_suffix("").name
    
        formatted_version = None if is_nightly else format_release_str(version)
    
        with open(file_path) as handle:
            # Strip and get non-empty lines
            parsed_file_names = [line.strip() for line in handle if line.strip()]
    
        # All non-nightly files have the date as their first line and they will all be on the same date
        if not is_nightly:
>           release_date = datetime.date.fromisoformat(parsed_file_names.pop(0))
E           AttributeError: type object 'datetime.date' has no attribute 'fromisoformat'

tools/impl/release_parsing.py:71: AttributeError
=========================================== short test summary info ===========================================
FAILED tools/tests/test_os_handling.py::test_invalid_codename_throws - Failed: DID NOT RAISE <class 'KeyError'>
FAILED tools/tests/test_release_parsing.py::test_mantid_releases - AttributeError: type object 'datetime.dat...
======================================== 2 failed, 37 passed in 0.15s =========================================
</details>


- Just a quick question for bonus points: Does [this Github action](https://github.com/mantidproject/download.mantidproject.org/actions/runs/2370407686/workflow#L5) actually run on the pull request branch itself, or on main??

--- a/tools/impl/os_handling.py
+++ b/tools/impl/os_handling.py
@@ -29,8 +29,8 @@ def get_osx_codename(filename: str) -> str:
     ]
 
     if not found_version:
+        # Assume that this is a conda-build that supports 10.9+
         return "10.9"
-        #raise KeyError(f"No known releases for {filename}")
 
     return found_version[-1]
 

--- a/tools/tests/test_os_handling.py
+++ b/tools/tests/test_os_handling.py
@@ -17,9 +17,8 @@ def test_between_high_sierra_and_sierra():
     assert os_handling.get_osx_codename("HighSierra") == "10.13"
 
 
-def test_invalid_codename_throws():
-    with pytest.raises(KeyError):
-        os_handling.get_osx_codename("Unknown")
+def test_missing_codename_returns_10_9():
+    assert os_handling.get_osx_codename("Unknown") == "10.9"
 
 
 @pytest.mark.parametrize("filename, expected",

